File: conteudo/Monografia/Sistema/API/src/main/java/mr/fmr/service/RepublicaService.java
package mr.fmr.service;

import mr.fmr.model.Estudante;
import mr.fmr.model.MoradorRepublica;
import mr.fmr.model.Republica;
import mr.fmr.model.User;

import java.security.Principal;
import java.util.List;

public interface RepublicaService {

    Republica save(User user);
    List<Republica> findAll();
    Republica findOne(long id);
    void delete(long id);
    List<Republica> findByCity(String city);
    List<Republica> findByUf(String uf);
    List<Republica> findByUfAndCity(String uf, String city);
    List<Estudante> findMoradores(Republica republica);
    List<Estudante> findPendentes(Republica republica);
    Republica createPersonality(Republica republica);
}

File: conteudo/Monografia/Sistema/FMR/src/app/service/republica.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { UrlService } from './default/url.service';

@Injectable({
  providedIn: 'root'
})
export class RepublicaService {

  constructor(private _http: HttpClient, private _url : UrlService) { }

  private baseUrl: string = this._url.baseUrl() + "/republica";

  findAll() {
    return this._http.get(this.baseUrl);
  }

  findOne(id) {
    return this._http.get(this.baseUrl + '/' + id);
  }

  findByCity(city) {
    return this._http.get(this.baseUrl + "/city/" + city);
  }

  findByUf(uf) {
    return this._http.get(this.baseUrl + "/uf/" + uf);
  }

  findByUfAndCity(uf, city) {
    return this._http.get(this.baseUrl + "/" + uf + "/" + city);
  }

  findMoradores() {
    return this._http.get(this.baseUrl + "/moradores");
  }

  findPendentes() {
    return this._http.get(this.baseUrl + "/pendentes")
  }

  aprovar(morador: any) {
    return this._http.post(this.baseUrl + "/aprovar", morador);
  }
  
  recusar(morador: any) {
    return this._http.post(this.baseUrl + "/recusar/" + morador, {});
  }

  remover(morador: any) {
    return this._http.post(this.baseUrl+"/remover/" + morador.id, {});
  }

  calcularPersonalidade(republica : any) {
    return this._http.post(this.baseUrl + "/personalidade", republica);
  }

  findBySearch(q) {
    return this._http.get(this.baseUrl + "/search/" + q.cidade + "/" + q.carater);
  }
}
